1-Draft/RFCNNNN-AdditionalTelemetry.md

```diff
@@ -0,0 +1,170 @@
+    User Voice
+  - Based on usage of this cmdlet there would potential for additional parameters
+- Disabling, and re-enabling telemetry will be available through a new cmdlet (Set-PSTelemetry):
+  - The cmdlet will stop all collection of telemetry for the remainder of that session and stop
+    telemetry from being collected on subsequent starts
+  - The user can request their telemetry is deleted by using a Remove-PSTelemetry cmdlet
+  - In order to have their telemetry deleted the user will be required to provide their System GUID
+    and User GUID
+  - If the cmdlet is unable to send the information the cmdlet will return a warning and alternative
+    instructions for support through GitHub. It will be specified that GUIDs are unique identifiers
+    and should not be shared publicly on GitHub.
+- There will be three options for the telemetry environment variable which can all be set manually
+  or using the Set-PSTelemetry cmdlet
+  - $True indicates all telemetry will be collected
+  - $False indicates no telemetry will be collected
+  - $NonIdentifying indicates that only non-identifying telemetry will be collected i.e. no GUIDs
+    will be collected
+- The existing disabling mechanisms for PowerShell telemetry will remain including the ability to
+  set the telemetry environment variable before the first start of PowerShell
+
+### Future work
+
+A public API for module authors and PowerShell hosted application owners to use the same
+infrastructure to collect telemetry for their module or application.
+Making the API public is a phase two goal of this project.
+We will make the telemetry publicly available once it is internally validated.
+Microsoft will not collect, store, or manage any data externally gathered through this API.
+
+## Design
+
+### PowerShell Changes
+
+- Count of unique devices will be based on a System GUID generated on the first start-up
+- Count of unique users will be based on a User GUID generated at PowerShell start time and stored
+  in the user's configuration directory
+- The combination of the User GUID and System GUID will be used to determine counts of unique
+  users and unique devices
+- The Application ID will be used to disambiguate PowerShell Hosted applications. It will be a
+  standardized method for applications to identify themselves in our telemetry. It will be up to
+  Application owners to tag their usage for inclusion in our counts.
+- The count execution type will be collected throughout the PowerShell session
+- The type of PowerShell session (hosted vs non-hosted) will be collected at PowerShell
+  engine start time
+- An example of what would be collected on PowerShell Start up:
+  - System GUID: b45d67e8-8c51-4ea3-a170-0a3cedb697ed
+  - User GUID: c58030d3-1a91-4086-9cb1-5bddd342056d
+  - The ApplicationID: VSCode
+  - The Platform being used: Windows 10 Pro
+  - The Version of PowerShell: 6.2.0-preview.4
+  - The type of PS session: Hosted

```

Comment:
According to the [Telemetry can only be disabled with an environment variable](https://docs.microsoft.com/en-us/powershell/scripting/whats-new/what-s-new-in-powershell-core-61?view=powershell-6#telemetry-can-only-be-disabled-with-an-environment-variable), PowerShell Core currently collects:

* OS name
* OS version
* PowerShell version

Might I suggest including that data for comparison here?

Reply:
I liked the idea of including a link to that documentation so I incorporated it into the above section